plugins/defaultplugins/l2plugin/bd_config.go

```diff
@@ -319,12 +319,29 @@ func (plugin *BDConfigurator) ResolveCreatedInterface(interfaceName string, inte
 	return nil
 }
 
-// ResolveDeletedInterface does nothing
-func (plugin *BDConfigurator) ResolveDeletedInterface(interfaceName string) {
+// ResolveDeletedInterface removes interface from bridge domain operational status
+func (plugin *BDConfigurator) ResolveDeletedInterface(interfaceName string) error {
 	log.DefaultLogger().Print("Interface was removed. Unregister from real state ", interfaceName)
+	// Lookup IfToBdIndexes in order to find a bridge domain for this interface
+	_, meta, found := plugin.IfToBdIndexes.LookupIdx(interfaceName)
+	if !found {
+		log.DefaultLogger().Debugf("Removed interface %s does not belong to any bridge domain", interfaceName)
+		return nil
+	}
+	bdID := meta.(*BridgeDomainMeta).BridgeDomainIndex
+	// Find bridge domain name
+	bdName, meta, found := plugin.BdIndexes.LookupName(bdID)
+	if !found {
+		return fmt.Errorf("unknown bridge domain ID %v", bdID)
+	}
+	err := plugin.LookupBridgeDomainDetails(bdID, bdName)
+	if err != nil {
+		return err
+	}
 	// Unregister removed interface
 	plugin.IfToBdRealStateIdx.UnregisterName(interfaceName)
-	// Nothing else to do here, vpp handles it itself
+
+	return nil
```

Comment:
is the error processed in watch_events.go?

Reply:
no, as well as for if, fib or xc. There is basically no error processing implemented in watch_events.go right now.

Reply:
Earlier that was not a problem (since this method returned no error effectively)